src/game/objects/common.c

```diff
@@ -44,13 +49,94 @@ const GAME_OBJECT_ID g_GunObjects[] = {
     NO_OBJECT,
 };
 
+const GAME_OBJECT_ID g_HoleObjects[] = {
+    O_KEY_HOLE1,    O_KEY_HOLE2,    O_KEY_HOLE3,    O_KEY_HOLE4,
+    O_PUZZLE_HOLE1, O_PUZZLE_HOLE2, O_PUZZLE_HOLE3, O_PUZZLE_HOLE4,
+    O_MIDAS_TOUCH,  NO_OBJECT,
+};
+
 const GAME_OBJECT_PAIR g_GunAmmoObjectMap[] = {
     { O_SHOTGUN_ITEM, O_SG_AMMO_ITEM },
     { O_MAGNUM_ITEM, O_MAG_AMMO_ITEM },
     { O_UZI_ITEM, O_UZI_AMMO_ITEM },
     { NO_OBJECT, NO_OBJECT },
 };
 
+const GAME_OBJECT_PAIR g_KeyItemToHoleMap[] = {
+    { O_KEY_OPTION1, O_KEY_HOLE1 },       { O_KEY_OPTION2, O_KEY_HOLE2 },
+    { O_KEY_OPTION3, O_KEY_HOLE3 },       { O_KEY_OPTION4, O_KEY_HOLE4 },
+    { O_PUZZLE_OPTION1, O_PUZZLE_HOLE1 }, { O_PUZZLE_OPTION2, O_PUZZLE_HOLE2 },
+    { O_PUZZLE_OPTION3, O_PUZZLE_HOLE3 }, { O_PUZZLE_OPTION4, O_PUZZLE_HOLE4 },
+    { O_LEADBAR_OPTION, O_MIDAS_TOUCH },  { NO_OBJECT, NO_OBJECT },
+};
+
+const GAME_OBJECT_BOUNDS_PAIR g_KeyItemHoleToBounds[] = {
+    { O_KEY_HOLE1, g_KeyHoleBounds },
+    { O_KEY_HOLE2, g_KeyHoleBounds },
+    { O_KEY_HOLE3, g_KeyHoleBounds },
+    { O_KEY_HOLE4, g_KeyHoleBounds },
+    { O_PUZZLE_HOLE1, g_PuzzleHoleBounds },
+    { O_PUZZLE_HOLE2, g_PuzzleHoleBounds },
+    { O_PUZZLE_HOLE3, g_PuzzleHoleBounds },
+    { O_PUZZLE_HOLE4, g_PuzzleHoleBounds },
+    { O_MIDAS_TOUCH, g_MidasBounds },
+    { NO_OBJECT, NULL },
+};
+
+GAME_OBJECT_ID Object_GetCognate(
+    GAME_OBJECT_ID key_id, const GAME_OBJECT_PAIR *test_map)
+{
+    const GAME_OBJECT_PAIR *pair = &test_map[0];
+    while (pair->key_id != NO_OBJECT) {
+        if (pair->key_id == key_id) {
+            return pair->value_id;
+        }
+        pair++;
+    }
+
+    return NO_OBJECT;
+}
+
+int16_t *Object_GetBoundsCognate(
+    GAME_OBJECT_ID key_id, const GAME_OBJECT_BOUNDS_PAIR *test_map)
+{
+    const GAME_OBJECT_BOUNDS_PAIR *pair = &test_map[0];
+    while (pair->key_id != NO_OBJECT) {
+        if (pair->key_id == key_id) {
+            // TODO Make all Lara_TestPosition use const bounds?
+            return (int16_t *)pair->bounds;
+        }
+        pair++;
+    }
+
+    return NULL;
+}
+
+int16_t Object_CheckForHole(GAME_OBJECT_ID object_id)
```

Comment:
Re the naming, perhaps `Object_FindKeyItemSlot`? "Key" in the sense it's a crucial item as opposed to just keys. Or perhaps `Object_FindReceptacle`.

Reply:
+1 for Object_FindReceptacle :)

Reply:
Love that name. I didn't want to use key since it was confusing on if it was key vs key item.